collect_app/src/main/java/org/odk/collect/android/projects/QrCodeProjectCreatorDialog.kt

```diff
@@ -128,16 +137,24 @@ class QrCodeProjectCreatorDialog : MaterialFullScreenDialogFragment() {
                 }
 
                 try {
-                    handleScanningResult(barcodeResult)
+                    val settingsJson = CompressionUtils.decompress(barcodeResult.text)
+                    lastScannedJson = settingsJson
+
+                    settingsUniquenessChecker.getProjectWithMatchingConnection(settingsJson)?.let { uuid ->
+                        lastMatchingUuid = uuid
+                        DialogUtils.showIfNotShowing(DuplicateProjectConfirmationDialog::class.java, childFragmentManager)
```

Comment:
You could pass the `lastScannedJson` and `lastMatchingUuid` into the dialog here using as args. `showIfNotShowing` can take args in as a `Bundle`.

Reply:
And then pass those values back on clicks.